extras/docker/fromsource/heketi-start.sh

```diff
@@ -1,5 +1,50 @@
 #!/bin/sh
 
+: ${HEKETI_PATH:=/var/lib/heketi}
+
+echo "Setting up heketi database"
+
+if [[ -d "${HEKETI_PATH}" ]]; then
+    # Test that our volume is writable.
+    touch "${HEKETI_PATH}/test" && rm "${HEKETI_PATH}/test"
+    if [ $? -ne 0 ]; then
+        echo "${HEKETI_PATH} is read-only"
+        exit 1
+    fi
+
+    if [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; then
+        echo "No database file found" | tee "${HEKETI_PATH}/container.log"
+        out=$(mount | grep "${HEKETI_PATH}" | grep heketidbstorage)
+        if [[ $? -eq 0 ]]; then
+            echo "Database volume found: ${out}" | tee "${HEKETI_PATH}/container.log"
+            echo "Database file is expected, waiting..." | tee "${HEKETI_PATH}/container.log"
+            check=0
+            while [[ ! -f "${HEKETI_PATH}/heketi.db" ]]; do
+                sleep 5
+                if [[ ${check} -eq 5 ]]; then
+                   echo "Database file did not appear, exiting." | tee "${HEKETI_PATH}/container.log"
+                   exit 1
+                fi
+                ((check+=1))
+            done
+        fi
+    fi
+
+    stat "${HEKETI_PATH}/heketi.db" | tee "${HEKETI_PATH}/container.log"
+    # Workaround for scenario where a lock on the heketi.db has not been
+    # released.
+    flock -w 60 "${HEKETI_PATH}/heketi.db" true
+    if [[ $? -ne 0 ]]; then
+        echo "Database file is read-only" | tee "${HEKETI_PATH}/container.log"
+    fi
+else
+    mkdir -p "${HEKETI_PATH}"
+    if [[ $? -ne 0 ]]; then
+        echo "Failed to create ${HEKETI_PATH}"
+        exit 1
+    fi
+fi
+
 if [ -f /backupdb/heketi.db.gz ] ; then
     gunzip -c /backupdb/heketi.db.gz > /var/lib/heketi/heketi.db
```

Comment:
`${HEKETI_PATH}` ?

Reply:
That's handled in the subsequent commit. The first one is purely about checking for a missing DB, the second is what touched the DB restore.

Reply:
Hm, but then it's slightly inconsistent in that you introduce the HEKETI_PATH variable, but only use it in the newly added lines. It' be more consistent to use it in all places? possibly introducing the variable in a preparatory commit?

But I won't block merging on this ...